xrdpvr/xrdpvr.c

```diff
@@ -176,27 +181,41 @@ xrdpvr_deinit_player(void *channel, int stream_id)
 int
 xrdpvr_play_media(void *channel, int stream_id, char *filename)
 {
-    int i;
+    int i, err;
 
     printf("$$$$$$ xrdpvr_play_media: setting audioTimeout & "
            "videoTimeout to -1\n");
     g_psi.videoTimeout = -1;
     g_psi.audioTimeout = -1;
 
-    /* register all available fileformats and codecs */
+    /* register all available fileformats and codecs
+     *
+     * FIXME -- this function is deprecated and eventually
+     * will be removed from the library. */
+#pragma GCC diagnostic push
```

Comment:
There's a few of these about. Presumably it's for compatibility with older versions of the library.  Is that right?

Reply:
Hi @matt335672,

> Suggest you remove the "for some strange reason". 

autoconf generates test files with the `.c` extention, and `g++` 11.3.1 on Fedora treats them as C language files, not as C++. So it looks strange for me that `g++` on Ubuntu doesn't do the same

> Presumably it's for compatibility with older versions of the library. Is that right?

Not exactly. These functions are marked as "deprecated" at the current version of the library, so they are likely to disappear in the reasonable future. Where possible, I've found a non-deprecated replacement, in another places I've just silenced the warning, but eventually these places must be somehow fixed.

Reply:
I'm not seeing that behaviour on a live Fedora 37 CD with g++ installed. On that platform, g++ will mangle external C names:-

```bash
$ cat temp.c
int externfunc();

int main()
{
    return externfunc();
}
$ gcc -c temp.c
$ nm temp.o | grep externfunc
                 U externfunc
$ g++ -c temp.c
$ nm temp.o | grep externfunc
                 U _Z10externfuncv
$ gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-redhat-linux/12/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-redhat-linux
Configured with: ../configure --enable-bootstrap --enable-languages=c,c++,fortran,objc,obj-c++,ada,go,d,lto --prefix=/usr --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=http://bugzilla.redhat.com/bugzilla --enable-shared --enable-threads=posix --enable-checking=release --enable-multilib --with-system-zlib --enable-__cxa_atexit --disable-libunwind-exceptions --enable-gnu-unique-object --enable-linker-build-id --with-gcc-major-version-only --enable-libstdcxx-backtrace --with-linker-hash-style=gnu --enable-plugin --enable-initfini-array --with-isl=/builddir/build/BUILD/gcc-12.2.1-20221121/obj-x86_64-redhat-linux/isl-install --enable-offload-targets=nvptx-none --without-cuda-driver --enable-offload-defaulted --enable-gnu-indirect-function --enable-cet --with-tune=generic --with-arch_32=i686 --build=x86_64-redhat-linux --with-build-config=bootstrap-lto --enable-link-serialization=1
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 12.2.1 20221121 (Red Hat 12.2.1-4) (GCC) 
```

As far as the "deprecated" warnings go, are you able to comment them with the version at which the deprecation warning was introduced? That will help all of us trying to maintain this going forwards.

Thanks.

Reply:
Hi @alexpevzner 

We seem to have stalled on this one. Given my comment above, are you happy to remove the "for some strange reason" and squash so we can get this one put to bed?

Thanks.